app/scripts/controllers/projects.js

```diff
@@ -8,6 +8,28 @@
  * Controller of the openshiftConsole
  */
 angular.module('openshiftConsole')
+  .filter('findProject', function() {
+    return function(items, searchString) {
+      var filtered = [];
+
+      if (searchString) {
+        var searchStringLower = searchString.toLowerCase();
+
+        _.forEach(items, function(item) {
+          if (item.metadata.name.toLowerCase().includes(searchStringLower) ||
```

Comment:
In [filters/resources.js](https://github.com/openshift/origin-web-console/blob/master/app/scripts/filters/resources.js#L17) we have a filter for getting at annotations that could be useful here.

```javascript
$filter('annotation')(item, 'displayName');
$filter('annotation')(item, 'description');
```